metadata-ingestion/src/datahub/specific/dataset.py

```diff
@@ -113,7 +114,7 @@ def add_upstream_lineage(self, upstream: Upstream) -> "DatasetPatchBuilder":
         self._add_patch(
             "upstreamLineage",
             "add",
-            path=f"/upstreams/{upstream.dataset}",
+            path=f"/upstreams/{quote(upstream.dataset, safe='')}",
```

Comment:
@hsheth2 This breaks integration tests as well but I had to do it as `s3` path with `/` caused lineage to break.

Reply:
Can we change this to exclusively quote `/` characters but not need to quote everything?

Also did you test how this works with our actual patch applier?